monitor/data_monitoring.py

```diff
@@ -26,20 +28,30 @@ class DataMonitoring(object):
     DBT_PROJECT_MODULES_PATH = os.path.join(DBT_PROJECT_PATH, 'dbt_modules', DBT_PACKAGE_NAME)
     DBT_PROJECT_PACKAGES_PATH = os.path.join(DBT_PROJECT_PATH, 'dbt_packages', DBT_PACKAGE_NAME)
 
-    def __init__(self, config: Config, force_update_dbt_package: bool = False,
-                 slack_webhook: Optional[str] = None) -> None:
+    def __init__(
+        self,
+        config: Config,
+        force_update_dbt_package: bool = False,
+        slack_webhook: Optional[str] = None, 
+        slack_token: Optional[str] = None,
+        slack_channel_name: Optional[str] = None
+    ) -> None:
         self.config = config
         self.dbt_runner = DbtRunner(self.DBT_PROJECT_PATH, self.config.profiles_dir, self.config.profile_target)
         self.execution_properties = {}
         self.slack_webhook = slack_webhook or self.config.slack_notification_webhook
+        self.slack_token = slack_token or self.config.slack_token
+        self.slack_channel_name = slack_channel_name or self.config.slack_notification_channel_name
+        # slack client is optional
+        self.slack_client = SlackClient.init(token=self.slack_token, webhook=self.slack_webhook) if (self.slack_token or self.slack_webhook) else None
```

Comment:
Why you don't provide the channel_name here to init as well? also - looking at the code and I was thinking that maybe init should be 'create_slack_client' as it's basically a factory method

Reply:
Will change the name :)